nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/export/DatasetExportServiceImpl.java

```diff
@@ -324,10 +324,10 @@ protected String notNullNxql(String nxql, List<PropertyType> featuresWithType) {
             String propName = prop.getName();
             if (IMAGE_TYPE.equals(prop.getType())) {
                 sb.append(" AND ").append(contentProperty(propName)).append(" IS NOT NULL");
-            } else if (CATEGORY_TYPE.equals(prop.getType())) {
+            } else if (CATEGORY_TYPE.equals(prop.getType()) || TEXT_TYPE.equals(prop.getType())) {
```

Comment:
This will be a problem When the type is text, and the property is a Blob to be converted. 
You need to find out what kind of property it is.

Reply:
Wizard allows to set any type to any property

Reply:
I know, that is why you need to check here if it is a string, or a blob.
In case of a blob, handle the same way as an image type

Reply:
There is an util function to check if it is a blob property